src/Nethermind/Nethermind.State/IReadOnlyStateProvider.cs

```diff
@@ -40,6 +40,8 @@ public interface IReadOnlyStateProvider
         
         Keccak GetCodeHash(Address address);
 
+        public bool IsContract(Address address) => GetCodeHash(address) != Keccak.OfAnEmptyString;
```

Comment:
this is generally not true, it is possible to deploy a contract with an empty code
There should be a method called HasCode?

Reply:
Can you call this contract later?